### DISCUSSION TITLE
Display problem of decemal in blockchain browser

### DISCUSSION THREAD

**Discussion by xuxinlai2002** [zkSync-Community-Hub/zksync-developers#245] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

We have released an erc20 contract for soul, allowing users to mint to obtain soul’s Token.

### What is the specific issue or error you're encountering?

The problem now is that we set decimals to 0, but the blockchain browser displays decimals (it seems to have set decimals to 1)

### Can you share the error messages or logs you're receiving, if any?

The contract address is 0x5155704BB41fDe152Ad3e1aE402e8E8b9bA335D3. After the contract is verified in the blockchain browser, the displayed decimals are 0
<img width="1121" alt="p1" src="https://github.com/zkSync-Community-Hub/zkync-developers/assets/29819239/9830c53c-f31d-42c7-971d-da7617c475af">
But in the transaction records, there are decimals displayed
<img width="1387" alt="P2" src="https://github.com/zkSync-Community-Hub/zkync-developers/assets/29819239/a36069b1-3399-49a6-bc78-d8aadc47740e">

### Have you made any recent changes to the contract before encountering this issue?

no

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

    function decimals() public view virtual override returns (uint8) {
        return 0;
    }

### Have you tried to isolate the problem, and if so, what were the results?

no

### What steps have you already taken to try to resolve the issue?

Consult zksync platform

**Comment by vasyl-ivanchuk** [comment, marked as answer]
Hi @xuxinlai2002 👋 , thanks for bringing it up. It seems that the Block Explorer UI doesn't take into consideration the changed implementation for a proxy contract. We will look into that.

**Comment by inpaslimabj** [comment]
It seems like there might be a discrepancy between the decimals specified in your contract code and how they are displayed in the blockchain browser. 

Verify that the contract's `decimals()` function is correctly returning 0, as you've specified. If the function is returning 0 as expected, then the discrepancy might be due to how the blockchain browser interprets and displays token information.

You may want to reach out to the developers or support team of the blockchain browser you're using to clarify this discrepancy and ensure accurate representation of your token's metadata.

**Comment by batesb188** [comment]
Encountering a discrepancy in decimal display for the ERC20 contract. Decimals set to 0 in code but displayed as 1 in transaction records. Seeking guidance from zkSync platform for resolution.

**Comment by lynnrosado** [comment]
It seems like there might be a discrepancy between the decimals set in your ERC20 contract and how it's displayed on the blockchain browser. You may want to double-check the settings in your contract and verify the contract source code to ensure consistency.

**Comment by tcaniao** [comment]
It seems there might be a discrepancy between how the decimals are displayed in different parts of the blockchain browser. 
You can try reaching out to the blockchain browser's support team for clarification or resolution.

**Comment by crissshawn** [comment]
The user has released an ERC20 contract for Soul, with decimals set to 0. However, the blockchain browser displays decimals as 1 in transaction records, despite showing 0 for the verified contract. The relevant portion of the contract code where the issue is occurring is the decimals() function, which returns 0. The user has not attempted to isolate the problem or taken any steps to resolve the issue, and has consulted the zkSync platform for assistance. No repository link was provided.

**Comment by eddiedenna** [comment]
The user has released an ERC20 contract for Soul and set the decimals to 0. However, the blockchain browser displays decimals as 1 in transaction records. The user has not made any recent changes to the contract and is not interacting with any external libraries or contracts. They have tried consulting the zkSync platform to resolve the issue and do not provide a repo link.
